sys/kern/sys_process.c

```diff
@@ -529,11 +529,11 @@ proc_read_cheri_tags(struct proc *p, struct uio *uio)
 	 * based on the tag bitmask size.
 	 */
 	va = uio->uio_offset;
-	if (!is_aligned(va, sizeof(uintcap_t)))
+	if (!is_aligned(va, sizeof(uintcap_t) * CHAR_BIT))
```

Comment:
Why is this change needed? The one below makes sense to me, but doesn't this restrict the VA to be a multiple of 128 bytes?

Reply:
Otherwise I need to do some bit-shifting nonsense to deal with misaligned tag fetches.  For example, if you fetch 8 tags starting at the second capability in a page, then I have to take the first two bytes of the tags returned from the page and synthesize the returned 8 bits by selecting 7 bits from the first byte and 1 bit from the second byte.  I didn't feel like solving that.

Reply:
Ah yes that makes sense. Might be worth adding a comment that the restriction is there for simplicity of implementation rather than fundamental.